core/src/test/java/org/elasticsearch/index/mapper/ip/IpFieldTypeTests.java

```diff
@@ -72,4 +72,51 @@ public void testTermQuery() {
         prefix = ip + "/16";
         assertEquals(InetAddressPoint.newPrefixQuery("field", InetAddresses.forString(ip), 16), ft.termQuery(prefix, null));
     }
+
+    public void testRangeQuery() {
+        MappedFieldType ft = createDefaultFieldType();
+        ft.setName("field");
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("::"),
+                        InetAddresses.forString("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")),
+                ft.rangeQuery(null, null, randomBoolean(), randomBoolean()));
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("::"),
+                        InetAddresses.forString("192.168.2.0")),
+                ft.rangeQuery(null, "192.168.2.0", randomBoolean(), true));
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("::"),
+                        InetAddresses.forString("192.168.1.255")),
+                ft.rangeQuery(null, "192.168.2.0", randomBoolean(), false));
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("2001:db8::"),
+                        InetAddresses.forString("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")),
+                ft.rangeQuery("2001:db8::", null, true, randomBoolean()));
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("2001:db8::1"),
+                        InetAddresses.forString("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")),
+                ft.rangeQuery("2001:db8::", null, false, randomBoolean()));
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("2001:db8::"),
+                        InetAddresses.forString("2001:db8::ffff")),
+                ft.rangeQuery("2001:db8::", "2001:db8::ffff", true, true));
+
+        assertEquals(
+                InetAddressPoint.newRangeQuery("field",
+                        InetAddresses.forString("2001:db8::1"),
+                        InetAddresses.forString("2001:db8::fffe")),
+                ft.rangeQuery("2001:db8::", "2001:db8::ffff", false, false));
+    }
```

Comment:
What about passing in the same value for upper and lower, and both inclusive=false. This should be an invalid range right? Do we have a check for that?